src/EventListener/AdminEventListener.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\EventListener;
+
+use Sonata\AdminBundle\Request\AdminFetcherInterface;
+use Sonata\AdminBundle\Templating\TemplateRegistryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\KernelEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Twig\Environment;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+final class AdminEventListener implements EventSubscriberInterface
+{
+    /**
+     * @var Environment
+     */
+    private $twig;
+
+    /**
+     * @var AdminFetcherInterface
+     */
+    private $adminFetcher;
+
+    /**
+     * @var TemplateRegistryInterface
+     */
+    private $templateRegistry;
+
+    public function __construct(
+        Environment $twig,
+        AdminFetcherInterface $adminFetcher,
+        TemplateRegistryInterface $templateRegistry
+    ) {
+        $this->twig = $twig;
+        $this->adminFetcher = $adminFetcher;
+        $this->templateRegistry = $templateRegistry;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => [['onKernelRequest', -50]],
```

Comment:
> > ## Subject
> > With this PR it is not necessary to use the `CRUDController::renderWithExtraParams` to render a template in the admin context. All admin pages with a `_sonata_admin` parameter inside the route configuration can use the autoinjected twig variables for that admin page.
> 
> What if the method was overridden by someone ?

This priority allows you to override the `base_template` variable in your custom controller.